### PR TITLE
Fixes bid resolution problems

### PR DESCRIPTION
**Pull request by ashfurrow**
This is a fix for a problem tracked in Trello: https://trello.com/c/Bio2jZQe/297-kiosk-showstopper-bid-polling-for-resolving-high-bidder-status-failing#

The issue is that bids weren't being placed (unless you were also registering, strangely). Actually this makes sense, once the underlying issue was found. 

We were sending `empty()` as a result of "we don't need to do this piece of work", and then trying to `flatMap` that empty observable to the bid-placing network call. But empty observables don't have anything to `flatMap` against, they're empty. So instead, now we're sending `just()`, which is short for `just(Void())`, which is a value (an empty one). So that value can get `flatMap`'d appropriately.

tl;dr we were returning an empty observable instead of an observable that sends an empty value.

Also, I fixed a problem where if you placed a bid, but it wasn't high enough, and then you hit "place higher bid", then changed your mind, the cancel button was gone and the app wouldn't let you leave without placing a higher bid.

### CODE REVIEW

File: Kiosk/Bid Fulfillment/LoadingViewController.swift [hunk 1]
```diff
@@ -142,6 +142,7 @@ extension LoadingViewController {
 
         let title = reserveNotMet ? "NO, THANKS" : (createdNewBidder ? "CONTINUE" : "BACK TO AUCTION")
         backToAuctionButton.setTitle(title, forState: .Normal)
+        fulfillmentContainer()?.cancelButton.hidden = false
```
Comment (acjay) [inline Kiosk/Bid Fulfillment/LoadingViewController.swift]: In a perfect world, would this attribute be hooked directly up to a signal?

File: Kiosk/Bid Fulfillment/BidderNetworkModel.swift [hunk 2]
```diff
@@ -110,7 +110,7 @@ private extension BidderNetworkModel {
             .filterSuccessfulStatusCodes()
             .map(void)
             .doOnCompleted { [weak self] in
-                // Adding the credit card succeeded, so we shoudl clear the newUser.creditCardToken so that we don't
+                // Adding the credit card succeeded, so we should clear the newUser.creditCardToken so that we don't
                 // inadvertently try to re-add their card token if they need to increase their bid.
 
                 self?.bidDetails.newUser.creditCardToken.value = nil
```
Comment (acjay) [inline Kiosk/Bid Fulfillment/BidderNetworkModel.swift]: I think this is one area where bugs tend to creep in -- we have mixes of FRP and of shared mutable state, which I think has the potential of dropping us to the lowest common denominator of reliability.

But not sure that's actionable right now, other than as a note to ourselves.

File: Kiosk/Bid Fulfillment/LoadingViewController.swift (hunk 1 above)
Comment (ashfurrow) [inline Kiosk/Bid Fulfillment/LoadingViewController.swift]: Yes, the whole `fulfillmentContainer()` is a poor abstraction that I've been meaning to replace when I get a chance. This added line is really fulfilling the promise in this comment: https://github.com/artsy/eidolon/blob/183210ea6c470e5819118ac0fda2d61fe5bc8a05/Kiosk/Bid%20Fulfillment/LoadingViewController.swift#L61

File: Kiosk/Bid Fulfillment/BidderNetworkModel.swift (hunk 2 above)
Comment (ashfurrow) [inline Kiosk/Bid Fulfillment/BidderNetworkModel.swift]: Agreed, something to be aware of as we move forward.

**Comment by acjay** [comment]
It looks good to go to me.

Regarding the original problem, it does seem to me that there's a semantic gap between the observable API and the UI/data flow. We may want to find ways to abstract over the subtleties of Rx, which may help prevent logic errors.

As alluded to in the comments, long term, I think we can improve upon a couple things architectural to go all in on FRP, but probably best to separate that from fixing the problem at hand.

Good fix!

**Comment by ashfurrow** [comment]
Thanks!

**Comment by acjay** [comment]
I'm still seeing Place Higher Bid when the Bid Submitted screen pops up. I think the polling may still be failing.

**Comment by ashfurrow** [comment]
It says the words "Bid Submitted"? Does it take a few seconds or does it happen immediately? I'll look into it.

**Comment by acjay** [comment]
It's after a delay, so I think it's timing out on the polling

**Comment by ashfurrow** [comment]
Hmm. Thanks.
